src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableActorContext.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Provides functionality for application code implementing an actor operation.
+    /// </summary>
+    public interface IDurableActorContext : IDeterministicExecutionContext
+    {
+        /// <summary>
+        /// Gets the class of the currently executing actor.
+        /// </summary>
+        string ActorClass { get; }
+
+        /// <summary>
+        /// Gets the key of the currently executing actor.
+        /// </summary>
+        string Key { get; }
+
+        /// <summary>
+        /// Gets an actor reference for the currently executing actor.
+        /// </summary>
+        ActorId Self { get; }
+
+        /// <summary>
+        /// Gets the name of the operation that was called.
+        /// </summary>
+        string OperationName { get; }
+
+        /// <summary>
+        /// A logger for logging information about this actor.
+        /// </summary>
+        ILogger Logger { get; }
+
+        /// <summary>
+        /// Whether this actor is freshly constructed, i.e. did not exist prior to this operation being called.
+        /// </summary>
+        bool NewlyConstructed { get; }
+
+        /// <summary>
+        /// Gets a typed view of the state, by deserializing the JSON.
+        /// </summary>
+        /// <typeparam name="TState">the JSON-serializable type of the actor state.</typeparam>
+        /// <returns>a typed view that allows reading and updating.</returns>
+        IStateView<TState> GetStateAs<TState>(Formatting formatting = Formatting.Indented, JsonSerializerSettings settings = null);
+
+        /// <summary>
+        /// Gets the content that was passed to this operation as a deserialized value.
+        /// </summary>
+        /// <typeparam name="T">The JSON-serializable type used for the operation content.</typeparam>
+        /// <returns>the operation content, or default(<typeparamref name="T"/>) if none.</returns>
+        T GetOperationContent<T>();
```

Comment:
I think we'll want to add a `<remarks>` paragraph explaining what exactly operation context is and how it is intended to be used. I don't even know what it is. :)

Reply:
I have added remarks to this method (as well as two others) that clarify the meaning of operation (name, content).